packages/headless/src/features/debug/ranking-info-parser.ts

```diff
@@ -1,44 +1,160 @@
-export type RankingInfo = ReturnType<typeof parseRankingInfo>;
+export interface RankingInformation {
+  /**
+   * The attributes of the document that contributed to its ranking.
+   * */
+  documentWeights: DocumentWeights | null;
 
-export interface ListOfWeights {
+  /**
+   * The weight attributed to each term in the query.
+   */
+  termsWeight: TermWeightReport | null;
+
+  /**
+   * The sum of all weights.
+   */
+  totalWeight: number | null;
+
+  /**
+   * The weights applied by query ranking expressions.
+   */
+  qreWeights: QueryRankingExpressionWeights[];
+}
+
+export interface DocumentWeights {
+  /**
+   * The effect of proximity of query terms. More weight is given to documents having the terms closer together.
+   */
   Adjacency: number;
+
+  /**
+   * The weight attributed to user ratings.
+   */
   'Collaborative rating': number;
+
+  /**
+   * The weight assigned through an [indexing pipeline extension (IPE)](https://docs.coveo.com/en/206/) for the item.
+   */
   Custom: number;
+
+  /**
+   * The weight attributed to the document date.
+   */
   Date: number;
+
+  /**
+   * The weight applied by a [query ranking expression (QRE)](https://docs.coveo.com/en/2777/coveo-solutions/using-query-ranking-expressions).
```

Comment:
Links to the doc site that have a slug don't need any of the semantic info that we inject after the slug. So this link would be the same as just https://docs.coveo.com/en/2777/ , and the ones just below would be https://docs.coveo.com/en/1624/#phase-2-item-weighting and https://docs.coveo.com/en/1448/ . It's not super important, but it keeps the links from being long and unwieldy.

Reply:
Nice tip, I will adjust,